web3/_utils/module_testing/eth_module.py

```diff
@@ -2672,6 +2694,37 @@ def test_eth_get_code_with_block_identifier(
         assert isinstance(code, HexBytes)
         assert len(code) > 0
 
+    def test_eth_create_access_list(
+        self,
+        w3: "Web3",
+        unlocked_account_dual_type: ChecksumAddress,
+        math_contract: "Contract",
+    ) -> None:
+        # Initialize transaction for gas estimation
+        txn_params: TxParams = {
+            "from": unlocked_account_dual_type,
+            "value": Wei(1),
+            "gas": 21000,
+        }
+
+        txn = math_contract.functions.incrementCounter(1).build_transaction(txn_params)
+
+        # create access list using data from transaction
+        response = w3.eth.create_access_list(
+            {
+                "from": unlocked_account_dual_type,
+                "to": math_contract.address,
+                "data": txn["data"],
+            }
+        )
+
+        assert is_dict(response)
+        access_list = response["accessList"]
+        assert len(access_list) > 0
+        assert access_list[0]["address"] is not None
```

Comment:
We should check the slots here too and that they are formatted appropriately.

Reply:
Formatting of the address I think is handled by the `is_checksum_address` on the next line. Is there another function I can use to check the formatting here?

Reply:
Well, we check the address but not the storage keys. We should check the storage keys formatting / values.

Reply:
Ah missed that, thanks for clarifying. 👍